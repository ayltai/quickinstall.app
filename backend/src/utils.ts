import { ErrorReporting, } from '@google-cloud/error-reporting';

const errors = new ErrorReporting();

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        errors.report(error);
    } else {
        console.error(error);
    }
};
