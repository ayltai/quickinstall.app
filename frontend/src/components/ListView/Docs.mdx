import { Controls, Description, Meta, Title, Primary, Stories, } from '@storybook/blocks';

import * as ComponentStories from './ListView.stories';

<Meta of={ComponentStories} />

<Title />

<Description />

# How to use lists

Lists can be used to help users find a specific item and act on it.

When using lists,
* Order list items in logical ways
* Keep items short and easy to scan
* Show icons, text, and actions in a consistent format

# Guidelines

* Start your item labels with a capital letter
* Don't use commas or any punctuation at the end of each item

# Default usage

<Primary />

# Parameters

<Controls sort='requiredFirst' />

<Stories
    title='Examples'
    includePrimary={false} />
