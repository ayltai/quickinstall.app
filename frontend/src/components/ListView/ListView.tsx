import DeleteForever from '@mui/icons-material/DeleteForever';
import Divider from '@mui/material/Divider';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import React, { Children, type FC, Fragment, } from 'react';

import type { ListViewProps, } from './ListView.types';

/**
 * A list displays one or more list items vertically.
 * @param children The content of the component
 * @param onClick Function to call when the list item is clicked
 * @param onDelete Function to call when the delete button is clicked
 * @param rest Other props
 */
export const ListView : FC<ListViewProps> = ({
    children,
    onClick,
    onDelete,
    ...rest
}) => (
    <List
        disablePadding
        {...rest}>
        {Children.toArray(children).map((child, index) => {
            const key = (child as any).key.substring(2);

            const handleClick = () => onClick && onClick(key);

            const handleDelete = () => onDelete && onDelete(key);

            return (
                <Fragment key={key}>
                    {index !== 0 && <Divider key={`divider-${key}`} />}
                    <ListItem
                        key={`listitem-${key}`}
                        disablePadding
                        secondaryAction={
                            <IconButton
                                data-testid='delete-button'
                                edge='end'
                                onClick={handleDelete}>
                                <DeleteForever />
                            </IconButton>
                        }>
                        <ListItemButton onClick={handleClick}>
                            {child}
                        </ListItemButton>
                    </ListItem>
                </Fragment>
            );
        })}
    </List>
);
