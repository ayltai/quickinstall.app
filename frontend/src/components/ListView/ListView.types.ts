import type { ListProps, } from '@mui/material/List';
import type { ReactNode, } from 'react';

export type ListViewProps = Omit<ListProps, 'children' | 'onClick'> & {
    children?  : ReactNode,
    onClick?   : (key : string) => void,
    onDelete?  : (key : string) => void,
};
