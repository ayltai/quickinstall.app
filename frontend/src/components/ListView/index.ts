export { ListView, } from './ListView';
export type { ListViewProps, } from './ListView.types';
