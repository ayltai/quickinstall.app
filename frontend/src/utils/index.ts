import * as Sentry from '@sentry/react';
import { FirebaseApp, } from 'firebase/app';
import { getAnalytics, logEvent, } from 'firebase/analytics';

export const handleError = (error : any) => {
    if (import.meta.env.PROD) {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const logPageView = (app : FirebaseApp, {
    pageName,
    route,
} : {
    pageName : string,
    route    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'page_view', {
            page_title    : pageName,
            page_location : route,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logAddToCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'add_to_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logRemoveFromCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'remove_from_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logCheckout = (app : FirebaseApp, {
    platform,
    itemIds,
} : {
    platform : 'win' | 'mac',
    itemIds  : string[],
}) => {
    try {
        logEvent(getAnalytics(app), 'checkout', {
            items : itemIds.map((itemId : string) => ({
                item_id : `${platform}/${itemId}`,
            })),
        });
    } catch (error) {
        handleError(error);
    }
};

export const logSearch = (app : FirebaseApp, {
    platform,
    query,
} : {
    platform : 'win' | 'mac',
    query    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'search', {
            search_term : `${platform}/${query}`,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logShare = (app : FirebaseApp, {
    product,
    itemId,
} : {
    product : 'win/packages' | 'mac/packages' | 'tweaks',
    itemId  : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'share', {
            content_type : product,
            item_id      : itemId,
        });
    } catch (error) {
        handleError(error);
    }
};
