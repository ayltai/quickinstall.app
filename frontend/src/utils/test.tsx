import { createTheme, ThemeProvider, } from '@mui/material';
import { configureStore, type Store, } from '@reduxjs/toolkit';
import { render, } from '@testing-library/react';
import React, { type ReactNode, } from 'react';
import { Provider, } from 'react-redux';
import { MemoryRouter, } from 'react-router-dom';

import { appStoreApi, chocolateyApi, contentfulApi, homebrewApi, quickInstallApi, } from '../apis';
import { recipeReducer, preferenceReducer, spellReducer, } from '../states';
import { appTheme, } from '../styles';

export const createStore = (preloadedState? : unknown) => configureStore({
    preloadedState,
    reducer    : {
        recipe                          : recipeReducer,
        preference                      : preferenceReducer,
        spell                           : spellReducer,
        [ appStoreApi.reducerPath  ]    : appStoreApi.reducer,
        [ chocolateyApi.reducerPath ]   : chocolateyApi.reducer,
        [ contentfulApi.reducerPath ]   : contentfulApi.reducer,
        [ homebrewApi.reducerPath  ]    : homebrewApi.reducer,
        [ quickInstallApi.reducerPath ] : quickInstallApi.reducer,
    },
    middleware : getDefaultMiddleware => getDefaultMiddleware().concat(appStoreApi.middleware, chocolateyApi.middleware, contentfulApi.middleware, homebrewApi.middleware, quickInstallApi.middleware),
});

export const defaultStore = createStore();

export type RootState = ReturnType<typeof defaultStore.getState>;

// @ts-ignore
const theme = createTheme(appTheme);

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : Record<string, any>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider>
            </MemoryRouter>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
