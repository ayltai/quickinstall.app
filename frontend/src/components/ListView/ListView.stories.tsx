import { ListItemText, } from '@mui/material';
import type { Meta, StoryObj, } from '@storybook/react';
import React from 'react';

import { ListView, } from './ListView';

const meta : Meta<typeof ListView> = {
    component : ListView,
    title     : 'Components/Layouts/List',
    argTypes  : {
        children : {
            description : 'The content of the component',
        },
        onClick  : {
            description : 'Function to call when the list item is clicked',
        },
        onDelete : {
            description : 'Function to call when the delete button is clicked',
        },
    },
};

export default meta;

type Story = StoryObj<typeof ListView>;

export const Default : Story = {
    args : {
        children : Array.from(Array(10)).map((_, index) => (
            <ListItemText
                key={String(Math.random())}
                primary={`Item ${index + 1}`}
                secondary={`Subtitle ${index + 1}`} />
        )),
    },
};
