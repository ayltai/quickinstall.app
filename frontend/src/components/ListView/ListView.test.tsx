import { ListItemText, } from '@mui/material';
import React from 'react';

import { fireEvent, render, } from '../../utils/test';
import { ListView, } from './ListView';

describe('<ListView />', () => {
    it('renders correctly', () => expect(render(
        <ListView>
            <ListItemText
                key='1'
                primary='Item 1'
                secondary='Item 1 description' />
            <ListItemText
                key='2'
                primary='Item 2'
                secondary='Item 2 description' />
            <ListItemText
                key='3'
                primary='Item 3'
                secondary='Item 3 description' />
        </ListView>
    )).toMatchSnapshot());

    it('triggers onClick when a list item is clicked', () => {
        const handleClick = vi.fn();

        const { getAllByRole, } = render(
            <ListView onClick={handleClick}>
                <ListItemText
                    key='1'
                    primary='Item 1'
                    secondary='Item 1 description' />
                <ListItemText
                    key='2'
                    primary='Item 2'
                    secondary='Item 2 description' />
                <ListItemText
                    key='3'
                    primary='Item 3'
                    secondary='Item 3 description' />
            </ListView>
        );

        fireEvent.click(getAllByRole('button')[0]);

        expect(handleClick).toHaveBeenCalledTimes(1);
        expect(handleClick).toHaveBeenCalledWith('1');
    });

    it('triggers onDelete when a delete button is clicked', () => {
        const handleDelete = vi.fn();

        const { getAllByTestId, } = render(
            <ListView onDelete={handleDelete}>
                <ListItemText
                    key='1'
                    primary='Item 1'
                    secondary='Item 1 description' />
                <ListItemText
                    key='2'
                    primary='Item 2'
                    secondary='Item 2 description' />
                <ListItemText
                    key='3'
                    primary='Item 3'
                    secondary='Item 3 description' />
            </ListView>
        );

        fireEvent.click(getAllByTestId('delete-button')[0]);

        expect(handleDelete).toHaveBeenCalledTimes(1);
        expect(handleDelete).toHaveBeenCalledWith('1');
    });
});
