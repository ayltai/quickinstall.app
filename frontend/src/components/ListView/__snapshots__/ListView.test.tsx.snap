// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ListView /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root css-1mk9mw3-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
        >
          <div
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
              >
                Item 1
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
              >
                Item 1 description
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              data-testid="delete-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteForeverIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
        >
          <div
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
              >
                Item 2
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
              >
                Item 2 description
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              data-testid="delete-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteForeverIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
        >
          <div
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
              >
                Item 3
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
              >
                Item 3 description
              </p>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              data-testid="delete-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteForeverIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root css-1mk9mw3-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
            >
              Item 1
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
            >
              Item 1 description
            </p>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            data-testid="delete-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <li
        class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
            >
              Item 2
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
            >
              Item 2 description
            </p>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            data-testid="delete-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <li
        class="MuiListItem-root MuiListItem-gutters css-5k8ysz-MuiListItem-root"
      >
        <div
          class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1hj7l4e-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1imhxo2-MuiTypography-root"
            >
              Item 3
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-4k7f0g-MuiTypography-root"
            >
              Item 3 description
            </p>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
            data-testid="delete-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteForeverIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
